src/redux/form/sagas/formDelete.saga.ts

```diff
@@ -20,23 +21,27 @@ export async function deleteFilm(film: Film) {
 }
 
 export function* deleteFilmTask(data: {
-  payload: FormPayload;
+  payload: Film;
 }): Generator<StrictEffect, void, any> {
-  const genre = data.payload.genre;
-  const searchTitle = data.payload.searchTitle;
+  const allMovies = yield select(allMoviesSelector);
+  const film = data.payload;
   try {
-    const film = data.payload.film;
-
     const res = yield call(deleteFilm, film);
 
     if (res) {
-      yield put(fetchfilmsList.request({ genre, searchTitle }));
+      const index = allMovies.indexOf(film);
```

Comment:
i don't think that searching object in array of objects with indexOf is a good way. It is probably will not work at all.
Search it by id value.